src/libtriton/includes/triton/bitwuzlaSolver.hpp

```diff
@@ -0,0 +1,125 @@
+//! \file
+/*
+**  Copyright (C) - Triton
+**
+**  This program is under the terms of the Apache License 2.0.
+*/
+
+#ifndef TRITON_BITWUZLASOLVER_H
+#define TRITON_BITWUZLASOLVER_H
+
+#include <chrono>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <bitwuzla/bitwuzla.h>
+
+#include <triton/ast.hpp>
+#include <triton/dllexport.hpp>
+#include <triton/solverEnums.hpp>
+#include <triton/solverInterface.hpp>
+#include <triton/solverModel.hpp>
+#include <triton/symbolicExpression.hpp>
+#include <triton/tritonTypes.hpp>
+
+
+//! The Triton namespace
+namespace triton {
+/*!
+ *  \addtogroup triton
+ *  @{
+ */
+  //! The Engines namespace
+  namespace engines {
+  /*!
+   *  \ingroup triton
+   *  \addtogroup engines
+   *  @{
+   */
+    //! The Solver namespace
+    namespace solver {
+    /*!
+     *  \ingroup engines
+     *  \addtogroup solver
+     *  @{
+     */
+
+      //! \class BitwuzlaSolver
+      /*! \brief Solver engine using Bitwuzla. */
+      class BitwuzlaSolver : public SolverInterface {
+        private:
+          /*! Struct used to provide information for Bitwuzla termination callback */
+          struct SolverParams {
+            SolverParams(int64_t timeout, size_t memory_limit):
+                timeout(timeout), memory_limit(memory_limit) {}
+
+            std::chrono::time_point<std::chrono::system_clock>
+              start = std::chrono::system_clock::now(); /*!< Solver starting time. */
+            int64_t timeout;             /*!< Timeout (ms) for solver instance running. */
+            size_t  memory_limit;        /*!< Memory limit for the whole symbolic process. */
+            size_t  call_cnt = 0;        /*!< Counter for the number of termination callback calls. */
+            int64_t last_mem_check = -1; /*!< Time when the last memory usage check was performed. */
+            size_t  delay = 1;           /*!< Check memory limit every delay seconds. */
+            size_t  code = 0;            /*!< Reason of solving termination (1 - timeout, 2 - out_of_mem). */
+          };
+
+        //! Callback function that implements termination of Bitwuzla solver on timeout and memory limit.
+        static int32_t terminate_fun(void* state);
```

Comment:
Is static needed (i think yes but just to be sure)? Maybe also renamed it to `terminate` or `terminateCallback` (to keep the correct coding style)?